pyani_plus/plot_run.py

```diff
@@ -110,68 +111,72 @@ def plot_heatmaps(
         sns.set_context("notebook", font_scale=scale)
 
     heatmaps_done = 0
-    for matrix, name, color_scheme in (
-        (run.identities, "identity", "spbnd_BuRd"),
-        (run.cov_query, "query_cov", "BuRd"),
-        (run.hadamard, "hadamard", "hadamard_BuRd"),
-    ):
-        if matrix is None:
-            # This is mainly for mypy to assert the matrix is not None
-            msg = f"ERROR: Could not load run {method} matrix"  # pragma: no cover
-            sys.exit(msg)  # pragma: no cover
-
-        nulls = int(matrix.isnull().sum().sum())
-        n = len(matrix)
-        if nulls:
-            msg = (
-                f"WARNING: Cannot plot {name} as matrix contains {nulls} nulls"
-                f" (out of {n}²={n**2} {method} comparisons)\n"
-            )
-            sys.stderr.write(msg)
-            continue
-
-        try:
-            matrix = run.relabelled_matrix(matrix, label)  # noqa: PLW2901
-        except ValueError as err:
-            msg = f"ERROR: {err}"
-            sys.exit(msg)
-
-        with warnings.catch_warnings():
-            warnings.filterwarnings(
-                "ignore",
-                message=(
-                    "scipy.cluster: The symmetric non-negative hollow observation"
-                    " matrix looks suspiciously like an uncondensed distance matrix"
-                ),
-            )
-            warnings.filterwarnings(
-                "ignore",
-                message=(
-                    "Clustering large matrix with scipy. Installing"
-                    " `fastcluster` may give better performance."
-                ),
-            )
-            figure = sns.clustermap(
-                matrix,
-                cmap=color_scheme,
-                vmin=0,
-                vmax=1,
-                figsize=(figsize, figsize),
-                linewidths=0.25,
-            )
+    with Progress(*PROGRESS_BAR_COLUMNS) as progress:
+        # Trailing space to match alignment of distributions progress bar:
+        task = progress.add_task("Plotting heatmaps...     ", total=3 * len(formats))
+        for matrix, name, color_scheme in (
+            (run.identities, "identity", "spbnd_BuRd"),
+            (run.cov_query, "query_cov", "BuRd"),
+            (run.hadamard, "hadamard", "hadamard_BuRd"),
```

Comment:
I think this can be refactored to load each matrix on demand (inside the loop, not before it).

Or, could be quicker to recalculate from identity * coverage, versus parsing serialised cached data-frame from the DB.